### ISSUE TITLE
factor_evaluation_solution.ipynb does not work

### ISSUE THREAD

**Issue by rydeen7** [stefan-jansen/alphalens-reloaded#2]
## Problem Description
I'm trying to run factor_evaluation_solution.ipynb in "Algorithmic Trading for Machine Learning".
But "create_summary_tear_sheet(factor_data)" return error.
 
**Please provide a minimal, self-contained, and reproducible example:**
```python

factor_data = utils.get_clean_factor_and_forward_returns(factor=factor,
                                                   prices=trade_prices,
                                                   quantiles=5,
                                                   max_loss=0.35,
                                                   periods=(1, 5, 10)).sort_index()


factor_data.info()

create_summary_tear_sheet(factor_data)
```

**Please provide the full traceback:**
```python

TypeError                                 Traceback (most recent call last)
<ipython-input-73-e56a52bd5b8f> in <module>
----> 1 create_summary_tear_sheet(factor_data)

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/alphalens/plotting.py in call_w_context(*args, **kwargs)
     44             with plotting_context(), axes_style(), color_palette:
     45                 sns.despine(left=True)
---> 46                 return func(*args, **kwargs)
     47         else:
     48             return func(*args, **kwargs)

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/alphalens/tears.py in create_summary_tear_sheet(factor_data, long_short, group_neutral)
    151     quantile_factor = factor_data["factor_quantile"]
    152 
--> 153     quantile_turnover = {
    154         p: pd.concat(
    155             [

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/alphalens/tears.py in <dictcomp>(.0)
    153     quantile_turnover = {
    154         p: pd.concat(
--> 155             [
    156                 perf.quantile_turnover(quantile_factor, q, p)
    157                 for q in range(1, int(quantile_factor.max()) + 1)

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/alphalens/tears.py in <listcomp>(.0)
    154         p: pd.concat(
    155             [
--> 156                 perf.quantile_turnover(quantile_factor, q, p)
    157                 for q in range(1, int(quantile_factor.max()) + 1)
    158             ],

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/alphalens/performance.py in quantile_turnover(quantile_factor, quantile, period)
    629     quant_turnover = (
    630         new_names.apply(lambda x: len(x))
--> 631         / quant_name_sets.apply(lambda x: len(x))
    632     ).rename(quantile)
    633     return quant_turnover

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/pandas/core/series.py in apply(self, func, convert_dtype, args, **kwds)
   4136             else:
   4137                 values = self.astype(object)._values
-> 4138                 mapped = lib.map_infer(values, f, convert=convert_dtype)
   4139 
   4140         if len(mapped) and isinstance(mapped[0], Series):

pandas/_libs/lib.pyx in pandas._libs.lib.map_infer()

~/opt/miniconda3/envs/ml4t/lib/python3.8/site-packages/alphalens/performance.py in <lambda>(x)
    629     quant_turnover = (
    630         new_names.apply(lambda x: len(x))
**--> 631         / quant_name_sets.apply(lambda x: len(x))**
    632     ).rename(quantile)
    633     return quant_turnover

**TypeError: object of type 'float' has no len()**

<Figure size 432x288 with 0 Axes>

```

**Please provide any additional information below:**

## Versions

* Alphalens version: 0.4.1.post1
* Python version: 3.8.8
* Pandas version: 1.2.4
* Matplotlib version: 3.4.1

**Comment by rydeen7** [comment]
I found that **get_clean_factor_and_forward_returns()**  usally returns Dataframe, but for some reason, it may return DataSeries. In addition, **quantile_factor** may be float.
I tried to fix this by myself, but so far unsuccessful.

Let me know how to fix.

**Comment by stefan-jansen** [comment]
This was fixed by release 0.4.2 September 2021.
